### PR TITLE
Initial support for Icom IC-7400

### PR DESCRIPTION
**Pull request by danielegozzi**
This should also work seamlessly for IC-746PRO, as it is using the same address for the CI-V protocol.

As with other existing icomciv.py based implementations, I did not add general radio settings: this implementation only allows for setting memories.

(Probably) related to #2157
I meant to mention WA1RKT for possibly trying out this on the IC-746PRO, but I'm not able to find him on GH. If this PR lands decently I can drop him an email.

# CHIRP PR Checklist

The following must be true before PRs can be merged:

* All tests must be passing.
* Commits should be squashed into logical units.
* Commits should be rebased (or simply rebase-able in the web UI) on current master. Do not put merge commits in a PR.
* Commits in a single PR should be related.
* Major new features or bug fixes should reference a [CHIRP issue](https://chirp.danplanet.com/projects/chirp/issues). Do this with the pattern `Fixes #1234` or `Related to #1234` so that the ticket system links the commit to the issue.
* New drivers should be accompanied by a test image in `tests/images` (except for thin aliases where the driver is sufficiently tested already).
* All files must be GPLv3 licensed or contain no license verbiage. No additional restrictions can be placed on the usage (i.e. such as noncommercial).

Please also follow these guidelines:

* Keep cleanups in separate commits from functional changes.
* Please write a reasonable commit message, especially if making some change that isn't totally obvious (such as adding a new model, adding a feature, etc).
* The first line of every commit is emailed to the users' list after each build. It should be short, but meaningful for regular users (examples: "thd74: Fixed tone decoding" or "uv5r: Added settings support").
* Do not add new py2-compatibility code (No new uses of `six`, `future`, etc).
* All new drivers should set `NEEDS_COMPAT_SERIAL=False` and use `MemoryMapBytes`.
* New drivers and radio models will affect the Python3 test matrix. You should regenerate this file with `tox -emakesupported` and include it in your commit.

### CODE REVIEW

File: chirp/drivers/icomciv.py [hunk 1]
```diff
@@ -900,6 +940,62 @@ def _initialize(self):
...
+        ]))
+        self._rf.valid_dtcs_codes = [
+            23, 25, 26, 31, 32, 36, 43, 47, 51, 53, 54, 65, 71, 72, 73, 74,
+            114, 115, 116, 122, 125, 131, 132, 134, 143, 145, 152, 155, 156,
+            162, 165, 172, 174, 205, 212, 223, 225, 226, 243, 244, 245, 246,
+            251, 252, 255, 261, 263, 265, 266, 271, 274, 306, 311, 315, 325,
+            331, 332, 343, 346, 351, 356, 364, 365, 371, 411, 412, 413, 423,
+            431, 432, 445, 446, 452, 454, 455, 462, 464, 465, 466, 503, 506,
+            516, 523, 526, 532, 546, 565, 606, 612, 624, 627, 631, 632, 654,
+            662, 664, 703, 712, 723, 731, 732, 734, 743, 754
```
[the remark this answers is not in the document]
Comment (danielegozzi) [reply chirp/drivers/icomciv.py:983]: Yes, my values are more specific and come directly from the IC-7400 manual (I cross-referenced them on the IC-746PRO manual too, and they are the same). The defaults from chirp_common contain more values, but those are not supported by the device.

File: chirp/drivers/icomciv.py [hunk 2]
```diff
@@ -900,6 +940,62 @@ def _initialize(self):
...
+        self._classes["mem"] = IC7400MemFrame
+        self._rf.has_bank = False
+        self._rf.has_dtcs_polarity = True
+        self._rf.has_dtcs = True
+        self._rf.has_ctone = True
+        self._rf.has_offset = False
+        self._rf.has_name = True
+        self._rf.has_tuning_step = False
+        self._rf.valid_modes = [
+            "LSB", "USB", "AM", "CW", "RTTY", "FM", "N/A", "CWR", "RTTYR"
```
[the remark this answers is not in the document]
Comment (danielegozzi) [reply chirp/drivers/icomciv.py:963]: Ok, I'll update that with USB. The outcome of repeating USB sounds good!

As I'm also trying to squash the commits, I'm not aware of a way of doing that without creating a new git history in a different branch. If I create a new branch with a single commit, you will also have to handle a new PR, right? If you are ok with that and we settle for keeping my DTCS/tones I can go on and try with a new overall commit.

[the remark this answers is not in the document]
Comment (kk7ds) [reply chirp/drivers/icomciv.py:963]: Nope, just squash them and `git push -f`. Don't create a new PR :)

**Comment by danielegozzi** [comment]
Suggestions implmented. The tones/dtcs codes were indeed the same as in chirp_common. Everything now squashed in one commit (obviously with a force push! how could I not think of that?).

**Comment by kk7ds** [comment]
Nice, thanks. I'll merge this for the next one.

Looks like there wasn't a bug for this one, so I filed it: https://www.chirpmyradio.com/issues/11246 I'll tweak your commit message to reference it (per item 5 on the checklist above) before I do.

**Pull request merged**